CRM/Civicase/Hook/PostProcess/AttachEmailActivityToAllCases.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/**
+ * Add bulk email as an activity to all the selected cases.
+ */
+class CRM_Civicase_Hook_PostProcess_AttachEmailActivityToAllCases {
+
+  /**
+   * Add bulk email as an activity to all the selected cases.
+   *
+   * @param string $formName
+   *   The class name of the submitted form.
+   * @param object $form
+   *   The submitted form instance.
+   */
+  public function run($formName, $form) {
+    if (!$this->shouldRun($formName)) {
+      return;
+    }
+
+    $this->addActivityToAllSelectedCases();
+  }
+
+  /**
+   * Add bulk email as an activity to all the selected cases.
+   */
+  public function addActivityToAllSelectedCases() {
+    $firstCaseId = CRM_Utils_Array::value('caseid', $_GET, '0');
+    $allCaseIds = array_diff(explode(',', CRM_Utils_Array::value('allCaseIds', $_GET, '0')),
+      [$firstCaseId]);
+    try {
+      $activity = civicrm_api3('Activity', 'get', [
+        'case_id' => $firstCaseId,
+        'options' => [
+          'limit' => 1,
+          'sort' => "id DESC",
+        ],
+        'sequential' => 1,
+        'return' => ['id'],
+      ]);
```

Comment:
This condition could be a problem. What if there was some race condition here and after all the emails got sent, another user created some bulk email with this same case `firstCaseid` as the first case and another activity was created right after the one for this email, that means that the activity for that will be picked instead of the one intended.

Can we add these parameter conditions the API
```php
 'activity_type_id' => 'Email',
  'source_contact_id' => "user_contact_id",
```

This ensures that the activity type will be email and that the user that created the activity is the logged in user.